src/v1_0/rws_client.cpp

```diff
@@ -250,20 +250,41 @@ std::string RWSClient::generateFilePath(const FileResource& resource)
   return Services::FILESERVICE + "/" + resource.directory + "/" + resource.filename;
 }
 
+// Try finding xml tag with class="code" containing retcode.
+std::optional<int> tryParseRetcode(Poco::AutoPtr<Poco::XML::Document> const& xml)
+{
+  std::string code_str = xmlFindTextContent(xml, XMLAttribute{"class", "code"});
+  try
+  {
+    return std::stoi(code_str);
+  }
+  catch (std::invalid_argument)
+  {
+    return {};
+  }
+  catch (std::out_of_range)
+  {
+    return {};
+  }
```

Comment:
This piece of code looks correct. I have two comments though:
1. If `xmlFindTextContent()` does not find the attribute, it returns an empty string. This will make `std::stoi()` throw, which is unnecessary (although the behavior will be correct).
2. The code can be made a bit shorted and multiple returns can be avoided by declaring a result variable.
```suggestion
  std::string code_str = xmlFindTextContent(xml, XMLAttribute{"class", "code"});
  std::optional<int> result;
  if (!code_str.empty())
  {
    try
    {
      result = std::stoi(code_str);
    }
    catch (std::invalid_argument) {}
    catch (std::out_of_range) {}
  }
  return result;
```